internal/helmdeployer/helmcache/secret.go

```diff
@@ -0,0 +1,76 @@
+package helmcache
+
+import (
+	"context"
+
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretClient struct {
+	cache     corecontrollers.SecretCache
+	client    kubernetes.Interface
+	namespace string
+}
+
+func NewSecretClient(cache corecontrollers.SecretCache, client kubernetes.Interface, namespace string) *SecretClient {
+	return &SecretClient{cache, client, namespace}
+}
+
+func (s *SecretClient) Create(ctx context.Context, secret *v1.Secret, opts metav1.CreateOptions) (result *v1.Secret, err error) {
+	return s.client.CoreV1().Secrets(s.namespace).Create(ctx, secret, opts)
```

Comment:
Are we sure that write operations (`Create`, `Update`, `Delete`, `DeleteCollection`, `Patch`, `Apply`?) write to the cache too? Is it worth testing here, or is this handled in Wrangler internals?
How the K8s client updates the cache is not clear to me here.

For instance:
* unit tests for the `List` operation already assume that the requested entries are in the cache
* those for `Create` don't expect any call to the cache
* `Create`/`Delete`, etc. test cases could check the state of the cache before/after the operation

Are these concerns relevant here, or should such behaviour be tested further up (eg. integration tests)?

Reply:
From my memory (and the PR description) we only want to cache Get/List. Maybe add function level comments to make it clear that's intentional.

Reply:
Added comments in https://github.com/rancher/fleet/pull/1809/commits/d9161ac38876582549e13a759ca8406691d7efae 

> How the K8s client updates the cache is not clear to me here.

This is handled by the wrangler `corev1controller.SecretCache`. You can read [wrangler](https://github.com/rancher/wrangler) and [lasso](https://github.com/rancher/lasso) docs to understand how they work. Also you can read about client-go [SharedInformers](https://aly.arriqaaq.com/kubernetes-informers/), which is what wrangler is using.

I don't think this is the right place to explain how cache works in wrangler, as we are using it extensively throughout the fleet code. In order to find where we are using wrangler caches just do a global search for `SecretCache`, `BundleCache`, `BundleDeploymentCache`, `ServiceAccountCache`, ...